src/main/kotlin/com/fasterxml/jackson/module/kotlin/KotlinModule.kt

```diff
@@ -104,6 +108,12 @@ internal class KotlinNamesAnnotationIntrospector(val module: KotlinModule, val c
     override fun findImplicitPropertyName(member: AnnotatedMember): String? {
         if (member is AnnotatedParameter) {
             return findKotlinParameterName(member)
+        } else if (member is AnnotatedMethod) {
+            if (member.declaringClass.isKotlinClass() && member.rawReturnType == Boolean::class.java) {
```

Comment:
Quick question: would this also need to take into account primitive `boolean` (`Boolean.TYPE`)?

Reply:
Add check Boolean as primitive

Reply:
So these changes are based on my branch originally so I wanted to provide some context for the boolean check. No, it is not strictly necessary but was done as a performance optimization to avoid the reflection lookups and field scanning that has to be done to find potential matching properties. Regarding catching all boolean types, that is admittedly an issue with my original PR that I addressed in a later PR (specifically also needing to handle Java booleans in additional to kotlin booleans which get mapped differently under certain circumstances): https://github.com/purecloudlabs/jackson-module-kotlin/pull/15/files.

Reply:
@DavidRigglemanININ thank you. I was wondering about that: although Kotlin spec does mention naming convention is for all "is-properties" it seems bit odd to use it for other types but then again people might use their own boolean-like types so maybe that's reasonable.

I am actually thinking of adding `jackson-databind` MapperFeature that would do this in databind itself (to be enabled by Kotlin module), and question of Boolean/boolean - or - wider would be valid question, still.